velox/vector/BaseVector.cpp

```diff
@@ -589,9 +589,17 @@ VectorPtr newConstant<TypeKind::LONG_DECIMAL>(
     variant& value,
     vector_size_t size,
     velox::memory::MemoryPool* pool) {
-  // LongDecimal variant is not supported to create
-  // constant vector.
-  VELOX_UNSUPPORTED();
+  // DECIMAL variant is not supported to create a ConstantVector.
```

Comment:
Shouldn't we fail if 'value' is not a null? It is misleading to return a null constant when the caller asked for a non-null constant.